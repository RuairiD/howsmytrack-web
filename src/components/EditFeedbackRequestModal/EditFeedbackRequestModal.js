import React from 'react';

import { Modal } from 'antd';
import EditFeedbackRequestForm from '../EditFeedbackRequestForm/EditFeedbackRequestForm';

type Props = {
    onCancel: () => void,
    isVisible: boolean,
    feedbackRequestId: number,
    soundcloudUrl: string,
    feedbackPrompt: string,
};

class FeedbackRequestModal extends React.Component<Props> {
    /*
     * Component for displaying modal for editing a feedback request.
     */
    render() {
        return (
            <Modal
                title="Edit Feedback Request"
                onCancel={this.props.onCancel}
                visible={this.props.isVisible}
                footer={null}
                destroyOnClose
            >
                <EditFeedbackRequestForm
                    feedbackRequestId={this.props.feedbackRequestId}
                    soundcloudUrl={this.props.soundcloudUrl}
                    feedbackPrompt={this.props.feedbackPrompt}
                />
            </Modal>
        );
    }
}

export default FeedbackRequestModal;
