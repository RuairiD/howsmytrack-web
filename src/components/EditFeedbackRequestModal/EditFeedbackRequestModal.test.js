import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import EditFeedbackRequestModal from "./EditFeedbackRequestModal";

describe("EditFeedbackRequestModal", () => {
    it("renders a Modal containing an EditFeedbackRequest", () => {
        const onCancel = () => {};
        const wrapper = shallow(
            <EditFeedbackRequestModal
                onCancel={onCancel}
                isVisible
                feedbackRequestId={1901}
                mediaUrl="https://soundcloud.com/ruairidx/sturzflug"
                feedbackPrompt="jery get ipad"
                emailWhenGrouped
                genre="electronic"
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
