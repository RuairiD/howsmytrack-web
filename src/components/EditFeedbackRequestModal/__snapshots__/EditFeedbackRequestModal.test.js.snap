// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFeedbackRequestModal renders a Modal containing an EditFeedbackRequest 1`] = `
<Modal
  className="responsive-modal"
  confirmLoading={false}
  destroyOnClose={true}
  footer={null}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="Edit Feedback Request"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form(UnwrappedEditFeedbackRequestForm)
    emailWhenGrouped={true}
    feedbackPrompt="jery get ipad"
    feedbackRequestId={1901}
    genre="electronic"
    mediaUrl="https://soundcloud.com/ruairidx/sturzflug"
  />
</Modal>
`;
